### PR TITLE
scylla_setup: add swapfile setup

### LINKED ISSUE #6539: Scylla_setup does warns about swap, but not help to set it correctly
Scylla version (or git commit hash): 4.0

Installing Scylla ion AWS i3 add the following warning:
**No swap is configured, it is highly recommended to setup swap on Scylla node.**

It is not clear what one should do next, if at all.
A better  UX will either:
- guide the user on fixing this 
- Add this as a step in scylla_setup (better)

### PR DESCRIPTION
**Pull request by syuu1228**
Adding swapfile setup on scylla_setup.

Fixes #6539

### CODE REVIEW

**Comment by slivne** [comment]
@noamha I know we had issue in the cloud with swap and setting it up - I can't recall all the incarnations

Can you please review this as well - at the end we would like the AMI to include SWAP as well.

Thanks

File: dist/common/scripts/scylla_swap_setup [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    parser.add_argument('--swap-directory',
+                        help='specify swapfile directory', required=True)
+    args = parser.parse_args()
+
+    if swap_exists():
+        print('swap already configured, exiting setup')
+        sys.exit(1)
+
+    memtotal = get_memtotal_gb()
+    swapsize = 16 if 16 < memtotal / 3 else memtotal / 3
```
Comment (avikivity) [inline dist/common/scripts/scylla_swap_setup:43]: This can be a huge amount. Let's clamp it between 16 and 50.

Comment (avikivity) [reply dist/common/scripts/scylla_swap_setup:43]: Better also convert it to an integer.

File: dist/common/scripts/scylla_swap_setup [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    swapfile = Path(args.swap_directory) / 'swapfile'
+    if swapfile.exists():
+        print('swapfile {} already exists'.format(swapfile))
+        sys.exit(1)
+    run('dd if=/dev/zero of={} bs=1GB count={}'.format(swapfile, swapsize))
+    swapfile.chmod(0o600)
+    run('mkswap -f {}'.format(swapfile))
+    run('swapon -v {}'.format(swapfile))
+    with open('/etc/fstab', 'a') as f:
+        f.write('{} swap swap defaults 0 0\n'.format(swapfile))
```
Comment (avikivity) [inline dist/common/scripts/scylla_swap_setup:58]: https://www.freedesktop.org/software/systemd/man/systemd.swap.html

File: dist/common/scripts/scylla_swap_setup (hunk 1 above)
Comment (syuu1228) [reply dist/common/scripts/scylla_swap_setup]: The logic is
```
if 16g <  (memtotal / 3)
    return 16g
else
    return (memtotal / 3)
```
It shouldn't be bigger than 16GB.
referenced: https://docs.scylladb.com/faq/#do-i-need-to-configure-swap-on-a-scylla-node

But agree with it need to to convert an integer, will fix.

**Comment by syuu1228** [comment]
@avikivity v2 pushed, please review again:
Changes from v1:
- Convert memtotal / 3 to int
- use fallocate instead of dd
- use systemd swap unit instead of fstab
- add default value of --swap-directory (default: /)

**Comment by slivne** [comment]
@noamha / @rjeczalik / @AdamNuclear  - I need the info how you create swap and any history of why 

We want to get swap creation in the ami so you guys can stop creating it.

File: dist/common/scripts/scylla_swap_setup [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        sys.exit(1)
+    run('fallocate -v -l {}GB {}'.format(swapsize, swapfile))
+    swapfile.chmod(0o600)
+    run('mkswap -f {}'.format(swapfile))
+    swapunit_bn = out('/usr/bin/systemd-escape -p --suffix=swap {}'.format(swapfile))
+    swapunit = Path('/etc/systemd/system/{}'.format(swapunit_bn))
+    if swapunit.exists():
+        print('swap unit {} already exists'.format(swapunit))
+        sys.exit(1)
+    unit_data = '''
```
Comment (avikivity) [inline dist/common/scripts/scylla_swap_setup:65]: You can use an f-string here: "unit_data = f```"

File: dist/common/scripts/scylla_swap_setup [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+[Unit]
+Description=swapfile
+
+[Swap]
+What={}
```
Comment (avikivity) [inline dist/common/scripts/scylla_swap_setup:70]: And then you can write `What={swapfile}` here, instead of the .format() call below.

Comment (avikivity) [reply dist/common/scripts/scylla_swap_setup:70]: This can be a follow-up.

**Comment by AdamNuclear** [comment]
@slivne I don't know history reasons but we are creating swap via this https://github.com/scylladb/siren/blob/master/command/cluster/create-swap.go

**Comment by noamha** [comment]
> @noamha / @rjeczalik / @AdamNuclear - I need the info how you create swap and any history of why
> 
> We want to get swap creation in the ami so you guys can stop creating it.

@slivne The issues we had with swap (at least int he current kernel version):
- Creating it on the root disk as a file caused corruption
- Creating it on a separate swap requires a reboot or else it created corruption on the data partition

**Pull request merged**